### PR TITLE
new method set_bot_commands

### PR DESCRIPTION
**Pull request by jonatan1609**
```python
app.set_bot_commands(
    [
         pyrogram.types.BotCommand("start", "start the bot"),
         pyrogram.types.BotCommand("stop", "stop the bot")
    ]
)
```
in order to remove all commands, we can pass `[]` or `None`.

### CODE REVIEW

File: pyrogram/types/bots_and_keyboards/bot_commands_list.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with Pyrogram.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import List
+
+from pyrogram import types
+from ..object import Object
+
+
+class BotCommandsList(Object):
```
Comment (delivrance) [inline pyrogram/types/bots_and_keyboards/bot_commands_list.py:25]: I don't think we need this. We can simply pass/return `List[BotCommand]`.

**Review by delivrance — CHANGES_REQUESTED** [review]

File: pyrogram/types/bots_and_keyboards/bot_commands_list.py (hunk 1 above)
Comment (jonatan1609) [reply pyrogram/types/bots_and_keyboards/bot_commands_list.py]: I think I've done that.

**Comment by jonatan1609** [comment]
I fixed a SyntaxError, thanks to Github Actions 😆

File: pyrogram/methods/bots/set_bot_commands.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+from pyrogram import types
+from pyrogram.scaffold import Scaffold
+
+
+class SetBotCommands(Scaffold):
+    async def set_bot_commands(
+            self,
+            commands: Union[
+                List[types.BotCommand],
+                List[List[str]],
```
Comment (delivrance) [inline pyrogram/methods/bots/set_bot_commands.py:31]: Here we can enforce the use of `List[types.BotCommand]` only to make it less confusing and also remove unnecessary checks in the code below.

**Review by delivrance — CHANGES_REQUESTED** [review]

File: pyrogram/methods/bots/set_bot_commands.py (hunk 2 above)
Comment (jonatan1609) [reply pyrogram/methods/bots/set_bot_commands.py]: Done

**Comment by delivrance** [comment]
I updated some parts since we don't need all that complexity.

**Pull request merged**